src/objective/lambdarank_obj.cc

```diff
@@ -225,10 +225,21 @@ class LambdaRankObj : public FitIntercept {
     };
 
     MakePairs(ctx_, iter, p_cache_, g, g_label, g_rank, loop);
-    if (sum_lambda > 0.0 && param_.lambdarank_normalization) {
-      double norm = std::log2(1.0 + sum_lambda) / sum_lambda;
-      std::transform(g_gpair.Values().data(), g_gpair.Values().data() + g_gpair.Size(),
-                     g_gpair.Values().data(), [norm](GradientPair const& g) { return g * norm; });
+    if (param_.lambdarank_normalization) {
+      double norm = 1.0;
+      if (param_.HasTruncation()) {
```

Comment:
How is this condition related to the use of the ``mean`` method? My understanding is that the normalization factor is ``n_pairs`` for the ``mean`` method and the gradient if other method is used?

From a quick reading of the source code, it seems like `HasTruncation()` is true if the LambdaRank pair method is ``topk``:

https://github.com/dmlc/xgboost/blob/9ab7567de4c6a197756c4726ee63966e38f33232/src/common/ranking_utils.h#L117

Are we pretty sure that there won't be any other pairing method besides ``topk`` and ``mean`` ?

Reply:
Any suggestions? Use a `switch` instead may help?

Reply:
Can we create a new method that returns ``true`` if the pair method is ``mean``? I think it's natural to have such a method because it requires a special logic for normalization.

Reply:
Done.